File: src/main/scala/org/xbery/artbeams/categories/domain/EditedCategory.scala
package org.xbery.artbeams.categories.domain

import java.util.Date

import net.formio.binding.ArgumentName
import org.xbery.artbeams.common.assets.domain.EditedTimeValidity

/**
  * @author Radek Beran
  */
case class EditedCategory(
  @ArgumentName("id")
  id: String,
  @ArgumentName("slug")
  slug: String,
  @ArgumentName("title")
  title: String,
  @ArgumentName("description")
  description: String,
  @ArgumentName("validFrom")
  override val validFrom: Date,
  @ArgumentName("validTo")
  override val validTo: Option[Date]
) extends EditedTimeValidity

File: src/main/kotlin/org/xbery/artbeams/categories/domain/EditedCategory.kt
package org.xbery.artbeams.categories.domain

import org.xbery.artbeams.common.assets.domain.EditedTimeValidity
import java.util.*

/**
 * @author Radek Beran
 */
data class EditedCategory(
    val id: String,
    val slug: String,
    val title: String,
    val description: String,
    override val validFrom: Date,
    override val validTo: Date?
) : EditedTimeValidity
